### PR TITLE
update the external links that contains master to release-5.0

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)
Update the external GitHub links that contain /master/ and /release-3.0/ to /release-5.0/.

### Which TiDB version(s) do your changes apply to? (Required)

- [ ] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/6054
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: hybrid-deployment-topology.md [hunk 1]
```diff
@@ -22,8 +22,8 @@ The deployment machine has multiple CPU processors with sufficient memory. To im
 
 ### Topology templates
 
-- [The simple template for the hybrid deployment](https://github.com/pingcap/docs-cn/blob/master/config-templates/simple-multi-instance.yaml)
-- [The complex template for the hybrid deployment](https://github.com/pingcap/docs/blob/master/config-templates/complex-multi-instance.yaml)
+- [The simple template for the hybrid deployment](https://github.com/pingcap/docs-cn/blob/release-5.0/config-templates/simple-multi-instance.yaml)
```
Comment (yikeke) [inline hybrid-deployment-topology.md:25]:
```suggestion
- [The simple template for the hybrid deployment](https://github.com/pingcap/docs/blob/release-5.0/config-templates/simple-multi-instance.yaml)
```
Please check if there are other similar mistakes

**Review by yikeke — APPROVED** [review]

File: hybrid-deployment-topology.md (hunk 1 above)
Comment (TomShawn) [reply hybrid-deployment-topology.md:25]: No other mistakes of this kind is found.

**Pull request merged**